src/utilities/service-utils.test.ts

```diff
@@ -1,13 +1,117 @@
 import { StubResourceRecord } from "../tests/stubs/stub-resource-record";
 import { Factory } from "rosie";
 import { AxiosResponse } from "axios";
 import { FactoryType } from "../tests/factories/factory-type";
 import { FactoryType as AndcultureCodeFactoryType } from "andculturecode-javascript-testing";
 import { ResultRecord } from "../view-models/result-record";
+import axios from "axios";
 import "jest-extended";
 
 describe("ServiceUtils", () => {
+    // -----------------------------------------------------------------------------------------
+    // #region configure
+    // -----------------------------------------------------------------------------------------
+
+    describe("configure", () => {
+        test.each`
+            cultureCode
+            ${null}
+            ${undefined}
+            ${""}
+            ${" "}
+        `(
+            "when cultureCode of $cultureCode, configures baseUrl with default culture code",
+            ({ cultureCode }) => {
+                // Arrange
+                const expected = LocalizationUtils.defaultCultureCode().toLowerCase();
+
+                // Act
+                ServiceUtils.configure(cultureCode);
+
+                // Assert
+                expect(axios.defaults.baseURL).toContain(expected);
+            }
+        );
+
+        test("when cultureCode set, configures baseUrl with provided value", () => {
+            // Arrange
+            const expected = faker.random.locale();
+
+            // Act
+            ServiceUtils.configure(expected);
+
+            // Assert
+            expect(axios.defaults.baseURL).toContain(expected.toLowerCase());
+        });
+
+        test("when API response handlers set, configures axio response interceptors", () => {
+            // Arrange
+            const expectedErrorHandler = (): any => {};
+            const expectedSuccessHandler = (): any => {};
+            const interceptorSpy = jest.spyOn(
+                axios.interceptors.response,
+                "use"
+            );
+
+            // Act
+            ServiceUtils.configure(
+                faker.random.locale(),
+                expectedErrorHandler,
+                expectedSuccessHandler
+            );
+
+            // Assert
+            expect(interceptorSpy).toHaveBeenCalledWith(
+                expectedSuccessHandler,
+                expectedErrorHandler
+            );
+        });
+    });
+
+    //#endregion configure
+
+    // -----------------------------------------------------------------------------------------
+    // #region configureCultureCode
+    // -----------------------------------------------------------------------------------------
+
+    describe("configureCultureCode", () => {
+        test.each`
+            cultureCode
+            ${null}
+            ${undefined}
+            ${""}
+            ${" "}
+        `(
+            "when cultureCode of $cultureCode, configures baseUrl with default culture code",
```

Comment:
Should these test names be "given cultureCode of $cultureCude, it configures baseUrl with default culture code"? Or is `when` more appropriate here? Seems to read weird, regardless of how close we're adhering to GWT naming.

Reply:
Yeah, definitely could be worded better. This may be one of those gray area scenarios, but I tend to use 'given' when its not a direct input of the literal system under tested method (something contextual) and 'when' its directly related to the action (or method being called)

At the end of the day, I 100% agree in rereading this it doesn't read well.

https://share.getcloudapp.com/9ZujRKoe

Reply:
@brandongregoryscott Thinking of our recent GWT convo in https://github.com/AndcultureCode/AndcultureCode.Cli/pull/70, aside from order (given > when) resulting in `_whenGivenOptions` becoming `_givenOptions`, in retrospect I suppose the word 'when' is _technically_ more appropriate, but (1) `_whenOptions` doesn't read well and (2) a CLI's optional flags felt more 'contextual' though honestly can very much be thought of as inputs. While that scenario had gray area it definitely need changed from `whenGiven`, but to what honestly I could go either way with `when` vs `given`.

Hoping that by sharing my train of thought when applying GWT between these two scenarios aids in better understanding it.

Reply:
Ohhh, gotcha - that makes more sense. I have definitely been merging them with "when given" for a while 😂 But in this scenario, I see how it differs - even though the data passed into the test is being arranged for the case, it's more contextual to the stsrem (and not being directly passed into the method under test.)